apps/fishing-map/features/datasets/upload/NewTrackDataset.tsx

```diff
@@ -156,22 +153,31 @@ function NewTrackDataset({
         ...(meta as DatasetMetadata),
         configuration: {
           ...meta?.configuration,
-          ...(newConfig as DatasetMetadata['configuration']),
+          configurationUI: {
+            ...meta?.configuration?.configurationUI,
+            ...(newConfig as DatasetMetadata['configuration']),
+          },
         },
       }))
     },
     []
   )
 
   const fieldsOptions: SelectOption[] = useMemo(
-    () => datasetMetadata?.fields?.map((field) => ({ id: field, label: field })) || [],
+    () =>
+      (datasetMetadata?.schema &&
+        Object.keys(datasetMetadata.schema).map((field) => ({ id: field, label: field }))) ||
```

Comment:
a very very nice to have feature could be to remove from this list the fields already selected

Reply:
Wouldn't we eventually want add more than one property to the filters array? I mean to add and remove items from the fields list? So we would need to have all available to toggle them on the list